test/unit/app/utils/test_kenya_state_machine.py

```diff
@@ -154,15 +163,72 @@ def test_change_language(mocker, test_client, init_database, session_factory, us
     state_machine.send_sms.assert_called_with("language_change_sms")
 
 
-# TODO: need pin authorisation implemented to do this
 def test_opt_out_of_marketplace(mocker, test_client, init_database):
     session = opt_out_of_market_place_pin_authorization_state()
     user = standard_user()
     user.phone = phone()
+    assert len(user.custom_attributes.all()) == 0
     state_machine = KenyaUssdStateMachine(session, user)
     state_machine.send_sms = mocker.MagicMock()
 
-    #state_machine.feed_char("0000")
-    #assert state_machine.state == "complete"
-    #assert user.custom_attributes.market_enabled == False
-    #state_machine.send_sms.assert_called_with("opt_out_of_market_place_sms")
+    state_machine.feed_char("0000")
+    assert state_machine.state == "complete"
+    market_enabled = list(filter(lambda x: x.name == 'market_enabled', user.custom_attributes.all()))
```

Comment:
@RubenGeo @enjeyw is this the correct way to get a specific custom_attribute...? seems very nonideal

Reply:
also the value ends up being str 'false' instead of bool False

Reply:
`user.custom_attributes` isn't actually a list of attributes, but rather an SQLAlchemy query, a better way of getting custom attributes is probably:
`user.custom_attributes.filter_by(name='market_enabled').all()`

Reply:
^ which is super efficient because it's converted into a raw postgres query.

Reply:
just tried setting the value type for the custom attributes to JSON on my local machine and fixes the typing issue nicely:
`

from server import db
from sqlalchemy.dialects.postgresql import JSON
from server.models.utils import ModelBase
from server.models.user import User


class CustomAttributeUserStorage(ModelBase):
    __tablename__ = 'custom_attribute_user_storage'

    name = db.Column(db.String)
    value = db.Column(JSON)

    user_id = db.Column(db.Integer, db.ForeignKey('user.id'))
    uploaded_image_id = db.Column(db.Integer)
`